idaes/power_generation/unit_models/tests/test_waterpipe.py

```diff
@@ -0,0 +1,119 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2020, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+Unit operation model for a waterpipe with pressure drop:
+
+Created on Aug 27, 2020 by Boiler Team (J. Ma, M. Zamarripa)
+"""
+import pytest
+# Import Pyomo libraries
+import pyomo.environ as pyo
+
+# Import IDAES core
+from idaes.core import FlowsheetBlock
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.power_generation.unit_models.waterpipe import WaterPipe
+# Import Unit Model Modules
+from idaes.generic_models.properties import iapws95
+
+from idaes.core.util.testing import get_default_solver, initialization_tester
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_default_solver()
+
+# -----------------------------------------------------------------------------
+
+
+@pytest.fixture(scope="module")
+def build_unit():
+    m = pyo.ConcreteModel()
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    m.fs.properties = iapws95.Iapws95ParameterBlock()
+    m.fs.unit = WaterPipe(default={
+                               "dynamic": False,
+                               "property_package": m.fs.properties,
+                               "has_holdup": True,
+                               "has_heat_transfer": False,
+                               "has_pressure_change": True,
+                               "water_phase": 'Liq',
+                               "contraction_expansion_at_end": 'contraction'
```

Comment:
It would be good to test the other cases (e.g. expansion in the pipe, plus vapor phase, to make sure the answers look right for those cases too.